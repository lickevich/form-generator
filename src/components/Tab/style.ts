import styled from "styled-components";
import { NavLink } from "react-router-dom";

export const StyledNavLink = styled(NavLink)`
  display: inline-block;
  padding: 10px 50px 5px;
  border: 1px solid black;
  border-bottom: none;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;

  &:first-child {
    border-right: none;
  }

  &:hover {
    text-decoration: underline;
  }

  &.selected {
    background-color: lightgray;
  }
`;
