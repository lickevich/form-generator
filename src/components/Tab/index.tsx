import React from "react";

import { StyledNavLink } from "./style";
import { ITab } from "../../type";

const Tab: React.FC<ITab> = ({ path, name }) => (
  <StyledNavLink exact to={path} activeClassName="selected">
    {name}
  </StyledNavLink>
);

export default Tab;
