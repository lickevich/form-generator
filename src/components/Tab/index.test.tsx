import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Tab from "./";

describe("Tab", () => {
  it("render the component", () => {
    const component = (
      <Router>
        <Tab path="/" name="Tab" />
      </Router>
    );
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
