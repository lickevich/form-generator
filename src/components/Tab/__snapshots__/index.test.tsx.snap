// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab render the component 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="sc-bdvvaa fojMIO selected"
    href="/"
  >
    Tab
  </a>
</DocumentFragment>
`;
